src/main/scala/mesosphere/marathon/core/group/impl/GroupManagerImpl.scala

```diff
@@ -45,7 +45,20 @@ class GroupManagerImpl(
     */
   private[this] val root = LockedVar(initialRoot)
 
-  override def rootGroup(): RootGroup = root.get()
+  @SuppressWarnings(Array("OptionGet"))
+  override def rootGroup(): RootGroup =
```

Comment:
What would you think if we had GroupManagerImpl queried to see if the post sync leadership has been completed, and then threw an exception if not? Ideas include potentially having the sync method complete some promise _before_ notifying anything else that the sync is complete